### ISSUE TITLE
Pure dict for plug-in data

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-base#117]
At the moment, data is accessed through methods on Context and Instance objects.

``` python
>>> context.set_data('key', 'value')
>>> context.data('key')
value
>>> context.has_data('key')
True
>>> context.remove_data('key')
```

Which are each the equivalent of pure dict functions, and could instead look like this.

``` python
>>> context.data['key'] = 'value'
>>> context.data['key']
'value'
>>> 'key' in context.data
True
>>> context.data.pop('key')
```

Which is a little more Pythonic. I'd vouch to save this change until the 1.2 as it is a breaking change that will require us all to re-factor our plug-ins and doesn't provide any additional benefit other than cosmetics.

**Comment by mottosso** [comment]
Implemented in 1.2.0

**Usage**

``` python
import pyblish.api as pyblish

context = pyblish.Context()

# Old behaviour preserved
context.set_data("key", "value")

# New behaviour preferred
context.data["key"] = "value"
if "key" in context.data:
  context.data.pop("key")

assert context.data.get("key") == None

# The same applies to Instances
instance = context.create_instance("MyInstance")
instance.data["key"] = "value"
```

The recommended way forwards is to start transitioning to using pure-dict data, not worrying too much about going back to refactor the old ways. Down the line, there will be a warning printed for each call to a deprecated member so as to easily find and refactor things.

From now on, guides and discussion will start to assume this new way of working.

**Comment by mottosso** [comment]
Implemented in [1.2.1](https://github.com/pyblish/pyblish/releases/tag/1.2.1).
